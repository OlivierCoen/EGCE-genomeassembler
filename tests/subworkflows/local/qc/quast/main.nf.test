nextflow_workflow {

    name "Test Workflow QC_QUAST"
    script "subworkflows/local/qc/quast/main.nf"
    workflow "QC_QUAST"
    tag "quast"

    test("One assembly") {

        tag "quast_one_assembly"

        setup {

            run("MAP_TO_REFERENCE") {
                script "subworkflows/local/map_to_reference/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [ id: "test", genome_size: 1500000 ],
                        file( '$projectDir/tests/test_data/pecat/input/reads.fasta.gz', checkIfExists: true)
                    ])
                    input[1] = Channel.of(
                        [
                            [ id: "test", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/primary.fasta', checkIfExists: true)
                        ]
                    )
                    """
                }
            }

        }

        when {

            workflow {
                """
                input[0] = MAP_TO_REFERENCE.out.aln_to_assembly_bam_ref
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Pecat output") {

        tag "quast_pecat_output"

        setup {

            run("MAP_TO_REFERENCE") {
                script "subworkflows/local/map_to_reference/main.nf"
                process {
                    """
                    input[0] = Channel.of(
                        [
                            [ id: "test", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/input/reads.small.fasta.gz', checkIfExists: true)
                        ]
                    )
                    input[1] = Channel.of(
                        [
                            [ id: "test", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/primary.fasta', checkIfExists: true)
                        ],
                        [
                            [ id: "test", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/alternate.fasta', checkIfExists: true)
                        ],
                        [
                            [ id: "test", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/haplotype_1.fasta', checkIfExists: true)
                        ],
                        [
                            [ id: "test", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/haplotype_2.fasta', checkIfExists: true)
                        ],
                        [
                            [ id: "test", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/rest_first_assembly.fasta', checkIfExists: true)
                        ],
                        [
                            [ id: "test", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/rest_second_assembly.fasta', checkIfExists: true)
                        ]
                    )
                    """
                }
            }

        }

        when {

            workflow {
                """
                input[0] = MAP_TO_REFERENCE.out.aln_to_assembly_bam_ref
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Multiple assemblies") {

        tag "quast_multiple_assemblies"

        setup {

            run("MAP_TO_REFERENCE") {
                script "subworkflows/local/map_to_reference/main.nf"
                process {
                    """
                    input[0] = Channel.of(
                        [
                            [ id: "test1", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/input/reads.fasta.gz', checkIfExists: true)
                        ],
                        [
                            [ id: "test2", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/input/reads.fasta.gz', checkIfExists: true)
                        ]
                    )
                    input[1] = Channel.of(
                        [
                            [ id: "test1", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/primary.fasta', checkIfExists: true)
                        ],
                        [
                            [ id: "test1", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/alternate.fasta', checkIfExists: true)
                        ],
                        [
                            [ id: "test2", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/primary.fasta', checkIfExists: true)
                        ],
                        [
                            [ id: "test2", genome_size: 1500000 ],
                            file( '$projectDir/tests/test_data/pecat/output/alternate.fasta', checkIfExists: true)
                        ]
                    )
                    """
                }
            }

        }

        when {

            workflow {
                """
                input[0] = MAP_TO_REFERENCE.out.aln_to_assembly_bam_ref
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
