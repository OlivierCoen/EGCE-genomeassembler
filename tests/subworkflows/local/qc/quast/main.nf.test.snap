{
    "Pecat output": {
        "content": [
            {
                "0": [
                    "test_report.tsv:md5,42273fbdd7ee7cf25f286e04d7115056"
                ],
                "1": [
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131"
                ],
                "quast_tsv": [
                    "test_report.tsv:md5,42273fbdd7ee7cf25f286e04d7115056"
                ],
                "versions": [
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-24T23:55:43.434315289"
    },
    "Multiple assemblies": {
        "content": [
            {
                "0": [
                    "test1_report.tsv:md5,17834a32abf232bdea0e363dc301f7fe",
                    "test1_report.tsv:md5,5a8b19657a571be8fc75aae21ebcdf1f",
                    "test2_report.tsv:md5,17834a32abf232bdea0e363dc301f7fe",
                    "test2_report.tsv:md5,5a8b19657a571be8fc75aae21ebcdf1f"
                ],
                "1": [
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131"
                ],
                "quast_tsv": [
                    "test1_report.tsv:md5,17834a32abf232bdea0e363dc301f7fe",
                    "test1_report.tsv:md5,5a8b19657a571be8fc75aae21ebcdf1f",
                    "test2_report.tsv:md5,17834a32abf232bdea0e363dc301f7fe",
                    "test2_report.tsv:md5,5a8b19657a571be8fc75aae21ebcdf1f"
                ],
                "versions": [
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-23T11:08:46.192943807"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "test_report.tsv:md5,615f60a4c98a9372c78b70ad00a74561",
                    "test_report.tsv:md5,a0b089a2d836aef9ea78c2187bc8cf39",
                    "test_report.tsv:md5,a0b6770c222ae4adcea6a499009d9374",
                    "test_report.tsv:md5,ba852487f1fd081ba9eb2a63fe487dff"
                ],
                "1": [
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131"
                ],
                "quast_tsv": [
                    "test_report.tsv:md5,615f60a4c98a9372c78b70ad00a74561",
                    "test_report.tsv:md5,a0b089a2d836aef9ea78c2187bc8cf39",
                    "test_report.tsv:md5,a0b6770c222ae4adcea6a499009d9374",
                    "test_report.tsv:md5,ba852487f1fd081ba9eb2a63fe487dff"
                ],
                "versions": [
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131",
                    "versions.yml:md5,6668e18804d1fcf9089d8ce396aab131"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-23T10:58:28.093796947"
    }
}